File: peractivity/src/main/java/spa/lyh/cn/peractivity/util/LanguageUtils.java
package spa.lyh.cn.peractivity.util;

import android.content.Context;

import java.lang.reflect.Method;

public class LanguageUtils {
    private final static String className = "spa.lyh.cn.languagepack.LanguagesPack";

    /**
     * 检查分享模块是否启动
     * @return
     */
    public static boolean isActivited(){
        try{
            Class.forName(className);
            return true;
        }catch (Exception ignored){
            return false;
        }
    }

    public static String getLanguageString(Context context,int id){
        Object obj;
        try{
            Class clazz = Class.forName(className);
            Method method = clazz.getMethod("getString",Context.class,int.class);
            obj = method.invoke(clazz,context,id);
            return (String) obj;
        }catch (Exception ignored){
            return "";
        }
    }
}

File: peractivity/src/main/java/spa/lyh/cn/peractivity/util/LanguageUtils.kt
package spa.lyh.cn.peractivity.util

import android.content.Context

object LanguageUtils {
    val className:String = "spa.lyh.cn.languagepack.LanguagesPack"

    @JvmStatic
    fun isActivited():Boolean{
        return try {
            Class.forName(className)
            true
        }catch (ignored:Exception){
            false
        }
    }

    @JvmStatic
    fun getLanguageString(context:Context,id:Int):String{
        val obj: Any
        return try {
            val clazz = Class.forName(className)
            val method = clazz.getMethod(
                "getString",
                Context::class.java,
                Int::class.javaPrimitiveType
            )
            obj = method.invoke(clazz, context, id) as Any
            (obj as String)
        } catch (ignored:Exception) {
            ""
        }
    }
}